src/test/feature-math.spec.ts

```diff
@@ -348,10 +348,51 @@ describe('feature-math', () => {
 
         for (let num = 1n; num <= 10n; num++) {
             for (let base = 2n; base <= 10; base++) {
-                const a = Math.floor(Math.log2(Number(num)));
-                const b = Math.floor(Math.log2(Number(base)));
-                const c = BigInt(Math.floor(a / b));
-                expect(await contract.getLog(num, base)).toBe(c);
+                const logarithm = BigInt(
+                    Math.floor(Math.log2(Number(num)) / Math.log2(Number(base)))
+                );
+                expect(await contract.getLog(num, base)).toBe(logarithm);
+            }
+        }
+
+        expect(await contract.getLog2(0n)).toBe(-1n);
+        expect(await contract.getLog(0n, 2n)).toBe(0n);
+
+        const maxint = 2n ** 256n - 1n;
+
+        function bigIntLogBase(num: bigint, base: bigint) {
+            let result = 0n;
+            while (num >= base) {
+                num = num / base;
+                result += 1n;
+            }
+            return result;
+        }
```

Comment:
This should be basically something like `Math.floor(Math.log2(Number(num)) / Math.log2(Number(base)))` as above. Otherwise the testing code just repeats the code being tested.